src/autopas/AutoPasImpl.h

```diff
@@ -0,0 +1,181 @@
+/**
+ * @file AutoPasImpl.h
+ * Contains all the missing implementations from AutoPasDecl.h that cannot be implemented there because AutoTuner and
+ * LogicHandler are only forward declared.
+ */
+
+#pragma once
+
+#include "autopas/AutoPasDecl.h"
+#include "autopas/LogicHandler.h"
+#include "autopas/selectors/AutoTuner.h"
+#include "autopas/selectors/tuningStrategy/TuningStrategyFactory.h"
```

Comment:
probably some includes missing. Here are some I spotted:
```suggestion
#include <memory>
#include <ostream>
#include <type_traits>

#include "autopas/AutoPasDecl.h"
#include "autopas/LogicHandler.h"
#include "autopas/selectors/AutoTuner.h"
#include "autopas/selectors/tuningStrategy/TuningStrategyFactory.h"
```

They are probably in `AutoPasDecl.h` and might be removable there.